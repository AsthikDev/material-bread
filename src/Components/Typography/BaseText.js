import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  default: {
    color: 'rgba(0, 0, 0, 0.87)',
  },
});

export default class BaseText extends Component {
  static propTypes = {
    style: PropTypes.object,
    children: PropTypes.node,
    typographyStyles: PropTypes.object,
    align: PropTypes.string,
    color: PropTypes.string,
    gutterBottom: PropTypes.bool,
  };

  render() {
    const {
      style,
      typographyStyles,
      align,
      color,
      gutterBottom,
      children,
    } = this.props;

    return (
      <Text
        style={[
          styles.default,
          typographyStyles,
          {
            textAlign: align ? align : 'left',
            color: color ? color : 'rgba(0,0,0, 87)',
            marginBottom: gutterBottom ? 10 : 0,
          },
          style,
        ]}
        {...this.props}>
        {children}
      </Text>
    );
  }
}
