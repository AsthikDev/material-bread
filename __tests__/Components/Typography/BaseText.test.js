import React from 'react';
import BaseText from '../../../src/Components/Typography/BaseText';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<BaseText />).toJSON();
  expect(tree).toMatchSnapshot();
});
