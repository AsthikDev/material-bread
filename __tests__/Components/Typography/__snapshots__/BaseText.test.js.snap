// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.87)",
    }
  }
>
  Text
</Text>
`;
